pybandits/strategy.py

```diff
@@ -20,15 +20,58 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
+from abc import ABC, abstractmethod
 from random import random
-from typing import Dict, List, Optional
+from typing import Any, Dict, List, Optional, Union
 
 import numpy as np
-from pydantic import validate_call
+from pydantic import field_validator, validate_call
 from scipy.stats import ttest_ind_from_stats
+from typing_extensions import Self
 
-from pybandits.base import ActionId, Float01, Model, Probability, Strategy
-from pybandits.model import Beta, BetaMOCC
+from pybandits.base import ActionId, Float01, Probability, PyBanditsBaseModel
+from pybandits.model import Beta, BetaMOCC, Model
+
+
+class Strategy(PyBanditsBaseModel, ABC):
+    """
+    Strategy to select actions in multi-armed bandits.
+    """
+
+    def _with_argument(self, argument_name: str, argument_value: Any) -> Self:
```

Comment:
I like this way to handle modification of the strategy params (avoiding set_method)